llama_hub/tools/neo4j_db/base.py

```diff
@@ -0,0 +1,146 @@
+from llama_index import Document
+from llama_index.graph_stores import Neo4jGraphStore
+from llama_index.tools.tool_spec.base import BaseToolSpec
+from llama_index.llms.base import LLM, ChatMessage
+
+
+class Neo4jQueryToolSpec(BaseToolSpec):
+    """
+    This class is responsible for querying a Neo4j graph database based on a provided schema definition.
+    """
```

Comment:
you need to define a `spec_functions` parameter defining the set of functions that will be available for the agent to use